symex-ts.el

```diff
@@ -0,0 +1,261 @@
+        (p (point)))
+    (symex-ts--get-topmost-node (tsc-get-named-descendant-for-position-range root p p))))
+
+
+(defun symex-ts-move-prev-sibling (&optional count)
+  "Move the point to the current node's previous sibling if possible.
+
+Move COUNT times, defaulting to 1."
+  (interactive "p")
+  (symex-ts--move-with-count #'tsc-get-prev-named-sibling '(-1 0) count))
+
+(defun symex-ts-move-next-sibling (&optional count)
+  "Move the point to the current node's next sibling if possible.
+
+Move COUNT times, defaulting to 1."
+  (interactive "p")
+  (symex-ts--move-with-count #'tsc-get-next-named-sibling '(1 0) count))
+
+(defun symex-ts-move-parent (&optional count)
+  "Move the point to the current node's parent if possible.
+
+Move COUNT times, defaulting to 1."
+  (interactive "p")
+  (symex-ts--move-with-count #'symex-ts--ascend-to-parent-with-sibling '(0 -1) count))
+
+(defun symex-ts-move-child (&optional count)
+  "Move the point to the current node's first child if possible.
+
+Move COUNT times, defaulting to 1."
+  (interactive "p")
+  (symex-ts--move-with-count #'symex-ts--descend-to-child-with-sibling '(0 1) count))
+
+(defun symex-ts-delete-node-backward (&optional count)
+  "Delete COUNT nodes backward from the current node."
+  (interactive "p")
+  (let* ((count (or count 1))
+         (node (symex-ts-get-current-node))
+         (end-pos (tsc-node-end-position node))
+         (start-pos (tsc-node-start-position
+                     (if (> count 1)
+                         (symex-ts--get-nth-sibling-from-node node #'tsc-get-prev-named-sibling count)
+                       node))))
+    (delete-region start-pos end-pos))
+  (symex-ts--after-tree-modification))
+
+(defun symex-ts-delete-node-forward (&optional count)
+  "Delete COUNT nodes forward from the current node."
+  (interactive "p")
+  (let* ((count (or count 1))
+         (node (symex-ts-get-current-node))
+         (start-pos (tsc-node-start-position node))
+         (end-pos (tsc-node-end-position
+                   (if (> count 1)
+                       (symex-ts--get-nth-sibling-from-node node #'tsc-get-next-named-sibling count)
+                     node))))
+    (delete-region start-pos end-pos))
+  (symex-ts--after-tree-modification))
+
+
+(defun symex-ts--hydra-exit ()
+  "Handle Hydra exit."
+  (symex-ts--delete-overlay))
+
+(defhydra hydra-symex-ts (:post (symex-ts--hydra-exit))
+  "Symex-TS."
+  ("d" symex-ts-current-node-sexp "DEBUG NODE")
+
+  ("h" symex-ts-move-prev-sibling "prev")
+  ("l" symex-ts-move-next-sibling "next")
+  ("j" symex-ts-move-parent "parent")
+  ("k" symex-ts-move-child "child")
+
+  ("X" symex-ts-delete-node-backward "delete node (backward)")
+  ("x" symex-ts-delete-node-forward "delete node (forward)"))
+
+(defun symex-ts-launch ()
+  "Start the Symex-TS hydra."
+  (interactive)
+
+  ;; Set the current node to the top-most node at point
+  (symex-ts-set-current-node-from-point)
+
+  ;; Launch hydra
+  (hydra-symex-ts/body))
```

Comment:
Since this hydra is a development tool rather than part of the functionality, I'd put this in [dev-utils.el](https://github.com/countvajhula/symex.el/blob/symex-ts-integration/dev/dev-utils.el)